{
  "uid" : "27bf3d3363484158",
  "name" : "Граничные значения для по количеству символов в имени категории, positive",
  "fullName" : "test.test_api.TestCategory#test_positive_boundary_values_for_name_category",
  "historyId" : "4e614238fb254857df90d7fce14a17a1",
  "time" : {
    "start" : 1750753066026,
    "stop" : 1750753066026,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nDETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\n\n[SQL: DELETE FROM category WHERE category.id = %(id)s]\n[parameters: {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)",
  "statusTrace" : "self = <sqlalchemy.engine.base.Connection object at 0x7fd71cfbf230>\ndialect = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7fd71cfbd7f0>\ncontext = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7fd71ce916e0>\nstatement = <sqlalchemy.dialects.postgresql.base.PGCompiler object at 0x7fd71ceb5bd0>\nparameters = [{'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\n\n    def _exec_single_context(\n        self,\n        dialect: Dialect,\n        context: ExecutionContext,\n        statement: Union[str, Compiled],\n        parameters: Optional[_AnyMultiExecuteParams],\n    ) -> CursorResult[Any]:\n        \"\"\"continue the _execute_context() method for a single DBAPI\n        cursor.execute() or cursor.executemany() call.\n    \n        \"\"\"\n        if dialect.bind_typing is BindTyping.SETINPUTSIZES:\n            generic_setinputsizes = context._prepare_set_input_sizes()\n    \n            if generic_setinputsizes:\n                try:\n                    dialect.do_set_input_sizes(\n                        context.cursor, generic_setinputsizes, context\n                    )\n                except BaseException as e:\n                    self._handle_dbapi_exception(\n                        e, str(statement), parameters, None, context\n                    )\n    \n        cursor, str_statement, parameters = (\n            context.cursor,\n            context.statement,\n            context.parameters,\n        )\n    \n        effective_parameters: Optional[_AnyExecuteParams]\n    \n        if not context.executemany:\n            effective_parameters = parameters[0]\n        else:\n            effective_parameters = parameters\n    \n        if self._has_events or self.engine._has_events:\n            for fn in self.dispatch.before_cursor_execute:\n                str_statement, effective_parameters = fn(\n                    self,\n                    cursor,\n                    str_statement,\n                    effective_parameters,\n                    context,\n                    context.executemany,\n                )\n    \n        if self._echo:\n            self._log_info(str_statement)\n    \n            stats = context._get_cache_stats()\n    \n            if not self.engine.hide_parameters:\n                self._log_info(\n                    \"[%s] %r\",\n                    stats,\n                    sql_util._repr_params(\n                        effective_parameters,\n                        batches=10,\n                        ismulti=context.executemany,\n                    ),\n                )\n            else:\n                self._log_info(\n                    \"[%s] [SQL parameters hidden due to hide_parameters=True]\",\n                    stats,\n                )\n    \n        evt_handled: bool = False\n        try:\n            if context.execute_style is ExecuteStyle.EXECUTEMANY:\n                effective_parameters = cast(\n                    \"_CoreMultiExecuteParams\", effective_parameters\n                )\n                if self.dialect._has_events:\n                    for fn in self.dialect.dispatch.do_executemany:\n                        if fn(\n                            cursor,\n                            str_statement,\n                            effective_parameters,\n                            context,\n                        ):\n                            evt_handled = True\n                            break\n                if not evt_handled:\n                    self.dialect.do_executemany(\n                        cursor,\n                        str_statement,\n                        effective_parameters,\n                        context,\n                    )\n            elif not effective_parameters and context.no_parameters:\n                if self.dialect._has_events:\n                    for fn in self.dialect.dispatch.do_execute_no_params:\n                        if fn(cursor, str_statement, context):\n                            evt_handled = True\n                            break\n                if not evt_handled:\n                    self.dialect.do_execute_no_params(\n                        cursor, str_statement, context\n                    )\n            else:\n                effective_parameters = cast(\n                    \"_CoreSingleExecuteParams\", effective_parameters\n                )\n                if self.dialect._has_events:\n                    for fn in self.dialect.dispatch.do_execute:\n                        if fn(\n                            cursor,\n                            str_statement,\n                            effective_parameters,\n                            context,\n                        ):\n                            evt_handled = True\n                            break\n                if not evt_handled:\n>                   self.dialect.do_execute(\n                        cursor, str_statement, effective_parameters, context\n                    )\n\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1964: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7fd71cfbd7f0>\ncursor = <cursor object at 0x7fd71c8a09a0; closed: -1>\nstatement = 'DELETE FROM category WHERE category.id = %(id)s'\nparameters = {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}\ncontext = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7fd71ce916e0>\n\n    def do_execute(self, cursor, statement, parameters, context=None):\n>       cursor.execute(statement, parameters)\nE       psycopg2.errors.ForeignKeyViolation: update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nE       DETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\n\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/default.py:945: ForeignKeyViolation\n\nThe above exception was the direct cause of the following exception:\n\ntmp_path_factory = TempPathFactory(_given_basetemp=PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw2'), _trace=<pluggy._tracing.TagTrac...4101490>, _basetemp=PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw2'), _retention_count=3, _retention_policy='all')\nworker_id = 'gw2'\nspends_client = <python_test.data_helper.api_helper.SpendsHttpClient object at 0x7fd71d0a5d30>\nspend_db = <python_test.databases.spend_db.SpendDb object at 0x7fd71d0a5160>\n\n    @pytest.fixture(scope=\"module\", autouse=True)\n    def module_fixture(tmp_path_factory, worker_id, spends_client: SpendsHttpClient, spend_db: SpendDb):\n        def _prepare_state():\n            all_spends_ids = spends_client.get_ids_all_spending()\n            for spend_id in all_spends_ids:\n                spends_client.delete_spending_by_id(spend_id)\n            categories_ids = spends_client.get_ids_all_categories()\n            spend_db.delete_categories_by_ids(categories_ids)\n    \n        if worker_id == \"master\":\n            _prepare_state()\n    \n        root_tmp_dir = tmp_path_factory.getbasetemp().parent\n    \n        fn = root_tmp_dir / \"prepare\"\n        with FileLock(str(fn) + \".lock\"):\n            if fn.is_file():\n                pass\n            else:\n>               _prepare_state()\n\ntest/test_api.py:40: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ntest/test_api.py:28: in _prepare_state\n    spend_db.delete_categories_by_ids(categories_ids)\ndatabases/spend_db.py:57: in delete_categories_by_ids\n    self.delete_category(category_id)\ndatabases/spend_db.py:52: in delete_category\n    session.commit()\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:2032: in commit\n    trans.commit(_to_root=True)\n<string>:2: in commit\n    ???\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py:139: in _go\n    ret_value = fn(self, *arg, **kw)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:1313: in commit\n    self._prepare_impl()\n<string>:2: in _prepare_impl\n    ???\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py:139: in _go\n    ret_value = fn(self, *arg, **kw)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:1288: in _prepare_impl\n    self.session.flush()\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:4353: in flush\n    self._flush(objects)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:4488: in _flush\n    with util.safe_reraise():\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/util/langhelpers.py:146: in __exit__\n    raise exc_value.with_traceback(exc_tb)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:4449: in _flush\n    flush_context.execute()\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py:466: in execute\n    rec.execute(self)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py:679: in execute\n    util.preloaded.orm_persistence.delete_obj(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py:193: in delete_obj\n    _emit_delete_statements(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py:1465: in _emit_delete_statements\n    c = connection.execute(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1416: in execute\n    return meth(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/sql/elements.py:523: in _execute_on_connection\n    return connection._execute_clauseelement(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1638: in _execute_clauseelement\n    ret = self._execute_context(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1843: in _execute_context\n    return self._exec_single_context(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1983: in _exec_single_context\n    self._handle_dbapi_exception(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:2352: in _handle_dbapi_exception\n    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1964: in _exec_single_context\n    self.dialect.do_execute(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7fd71cfbd7f0>\ncursor = <cursor object at 0x7fd71c8a09a0; closed: -1>\nstatement = 'DELETE FROM category WHERE category.id = %(id)s'\nparameters = {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}\ncontext = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7fd71ce916e0>\n\n    def do_execute(self, cursor, statement, parameters, context=None):\n>       cursor.execute(statement, parameters)\nE       sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nE       DETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\nE       \nE       [SQL: DELETE FROM category WHERE category.id = %(id)s]\nE       [parameters: {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\nE       (Background on this error at: https://sqlalche.me/e/20/gkpj)\n\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/default.py:945: IntegrityError",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "[S] Auth Token",
    "time" : {
      "start" : 1750753066191,
      "stop" : 1750753066545,
      "duration" : 354
    },
    "status" : "passed",
    "steps" : [ {
      "name" : "GET /oauth2/authorize",
      "time" : {
        "start" : 1750753066211,
        "stop" : 1750753066265,
        "duration" : 54
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ {
        "uid" : "14f8fc90cfd8bbe9",
        "name" : "Request",
        "source" : "14f8fc90cfd8bbe9.html",
        "type" : "text/html",
        "size" : 2119
      }, {
        "uid" : "66eda35ab733b54d",
        "name" : "Response text 200",
        "source" : "66eda35ab733b54d.txt",
        "type" : "text/plain",
        "size" : 2897
      }, {
        "uid" : "7270eb0d741bd6c0",
        "name" : "Response headers 200",
        "source" : "7270eb0d741bd6c0.json",
        "type" : "application/json",
        "size" : 667
      } ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 3,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    }, {
      "name" : "POST /login",
      "time" : {
        "start" : 1750753066273,
        "stop" : 1750753066436,
        "duration" : 163
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ {
        "uid" : "baddbffe987ce337",
        "name" : "Request",
        "source" : "baddbffe987ce337.html",
        "type" : "text/html",
        "size" : 2397
      }, {
        "uid" : "c9b3a6ac1dc8a9cc",
        "name" : "Response text 200",
        "source" : "c9b3a6ac1dc8a9cc.txt",
        "type" : "text/plain",
        "size" : 417
      }, {
        "uid" : "27033a9852445d87",
        "name" : "Response headers 200",
        "source" : "27033a9852445d87.json",
        "type" : "application/json",
        "size" : 287
      } ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 3,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    }, {
      "name" : "POST /oauth2/token",
      "time" : {
        "start" : 1750753066444,
        "stop" : 1750753066545,
        "duration" : 101
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ {
        "uid" : "62d289ae8377ff6",
        "name" : "Request",
        "source" : "62d289ae8377ff6.html",
        "type" : "text/html",
        "size" : 3376
      }, {
        "uid" : "412aab4b14b103e7",
        "name" : "Response json 200",
        "source" : "412aab4b14b103e7.json",
        "type" : "application/json",
        "size" : 1529
      }, {
        "uid" : "d0f766dcc2710862",
        "name" : "Response headers 200",
        "source" : "d0f766dcc2710862.json",
        "type" : "application/json",
        "size" : 504
      } ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 3,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 3,
    "attachmentsCount" : 9,
    "shouldDisplayMessage" : false,
    "hasContent" : true
  }, {
    "name" : "[S] App User",
    "time" : {
      "start" : 1750753066030,
      "stop" : 1750753066191,
      "duration" : 161
    },
    "status" : "passed",
    "steps" : [ {
      "name" : "Создать пользователя 'qa_guru'",
      "time" : {
        "start" : 1750753066031,
        "stop" : 1750753066190,
        "duration" : 159
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "user_name",
        "value" : "'qa_guru'"
      }, {
        "name" : "user_password",
        "value" : "'qa_guru'"
      } ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 1,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : true
  }, {
    "name" : "[S] Envs",
    "time" : {
      "start" : 1750753066030,
      "stop" : 1750753066030,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "[S] Tmp Path Factory",
    "time" : {
      "start" : 1750753066545,
      "stop" : 1750753066545,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "[M] Module Fixture",
    "time" : {
      "start" : 1750753066659,
      "stop" : 1750753066988,
      "duration" : 329
    },
    "status" : "broken",
    "statusMessage" : "sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nDETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\n\n[SQL: DELETE FROM category WHERE category.id = %(id)s]\n[parameters: {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)\n",
    "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n          ^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n            ^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 139, in _multicall\n    teardown.throw(exception)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 53, in run_old_style_hookwrapper\n    return result.get_result()\n           ~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_result.py\", line 103, in get_result\n    raise exc.with_traceback(tb)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 38, in run_old_style_hookwrapper\n    res = yield\n          ^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/pluggy/_callers.py\", line 121, in _multicall\n    res = hook_impl.function(*args)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/fixtures.py\", line 1140, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/_pytest/fixtures.py\", line 898, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/niffler-py-st3/niffler-py-st3/python_test/test/test_api.py\", line 40, in module_fixture\n    _prepare_state()\n    ~~~~~~~~~~~~~~^^\n  File \"/home/runner/work/niffler-py-st3/niffler-py-st3/python_test/test/test_api.py\", line 28, in _prepare_state\n    spend_db.delete_categories_by_ids(categories_ids)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/allure_commons/_allure.py\", line 202, in impl\n    return func(*a, **kw)\n  File \"/home/runner/work/niffler-py-st3/niffler-py-st3/python_test/databases/spend_db.py\", line 57, in delete_categories_by_ids\n    self.delete_category(category_id)\n    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/allure_commons/_allure.py\", line 202, in impl\n    return func(*a, **kw)\n  File \"/home/runner/work/niffler-py-st3/niffler-py-st3/python_test/databases/spend_db.py\", line 52, in delete_category\n    session.commit()\n    ~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 2032, in commit\n    trans.commit(_to_root=True)\n    ~~~~~~~~~~~~^^^^^^^^^^^^^^^\n  File \"<string>\", line 2, in commit\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py\", line 139, in _go\n    ret_value = fn(self, *arg, **kw)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 1313, in commit\n    self._prepare_impl()\n    ~~~~~~~~~~~~~~~~~~^^\n  File \"<string>\", line 2, in _prepare_impl\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py\", line 139, in _go\n    ret_value = fn(self, *arg, **kw)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 1288, in _prepare_impl\n    self.session.flush()\n    ~~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4353, in flush\n    self._flush(objects)\n    ~~~~~~~~~~~^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4488, in _flush\n    with util.safe_reraise():\n         ~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/util/langhelpers.py\", line 146, in __exit__\n    raise exc_value.with_traceback(exc_tb)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4449, in _flush\n    flush_context.execute()\n    ~~~~~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py\", line 466, in execute\n    rec.execute(self)\n    ~~~~~~~~~~~^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py\", line 679, in execute\n    util.preloaded.orm_persistence.delete_obj(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        self.mapper,\n        ^^^^^^^^^^^^\n        uow.states_for_mapper_hierarchy(self.mapper, True, False),\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n        uow,\n        ^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py\", line 193, in delete_obj\n    _emit_delete_statements(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        base_mapper,\n        ^^^^^^^^^^^^\n    ...<3 lines>...\n        delete,\n        ^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py\", line 1465, in _emit_delete_statements\n    c = connection.execute(\n        statement, del_objects, execution_options=execution_options\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1416, in execute\n    return meth(\n        self,\n        distilled_parameters,\n        execution_options or NO_OPTIONS,\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/sql/elements.py\", line 523, in _execute_on_connection\n    return connection._execute_clauseelement(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        self, distilled_params, execution_options\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1638, in _execute_clauseelement\n    ret = self._execute_context(\n        dialect,\n    ...<8 lines>...\n        cache_hit=cache_hit,\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1843, in _execute_context\n    return self._exec_single_context(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~^\n        dialect, context, statement, parameters\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1983, in _exec_single_context\n    self._handle_dbapi_exception(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        e, str_statement, effective_parameters, cursor, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 2352, in _handle_dbapi_exception\n    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1964, in _exec_single_context\n    self.dialect.do_execute(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        cursor, str_statement, effective_parameters, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/default.py\", line 945, in do_execute\n    cursor.execute(statement, parameters)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^\n",
    "steps" : [ {
      "name" : "Получить id всех трат пользователя",
      "time" : {
        "start" : 1750753066762,
        "stop" : 1750753066884,
        "duration" : 122
      },
      "status" : "passed",
      "steps" : [ {
        "name" : "GET /api/spends/all?filterCurrency=RUB",
        "time" : {
          "start" : 1750753066770,
          "stop" : 1750753066789,
          "duration" : 19
        },
        "status" : "passed",
        "steps" : [ ],
        "attachments" : [ {
          "uid" : "264bc1a9bff3dd36",
          "name" : "Request",
          "source" : "264bc1a9bff3dd36.html",
          "type" : "text/html",
          "size" : 3623
        }, {
          "uid" : "2431792203945392",
          "name" : "Response json 200",
          "source" : "2431792203945392.json",
          "type" : "application/json",
          "size" : 2
        }, {
          "uid" : "5ca035cf04db5287",
          "name" : "Response headers 200",
          "source" : "5ca035cf04db5287.json",
          "type" : "application/json",
          "size" : 490
        } ],
        "parameters" : [ ],
        "stepsCount" : 0,
        "attachmentsCount" : 3,
        "shouldDisplayMessage" : false,
        "hasContent" : true
      }, {
        "name" : "GET /api/spends/all?filterCurrency=KZT",
        "time" : {
          "start" : 1750753066795,
          "stop" : 1750753066816,
          "duration" : 21
        },
        "status" : "passed",
        "steps" : [ ],
        "attachments" : [ {
          "uid" : "287aadaf9ba47d4d",
          "name" : "Request",
          "source" : "287aadaf9ba47d4d.html",
          "type" : "text/html",
          "size" : 3623
        }, {
          "uid" : "40766f9915552038",
          "name" : "Response json 200",
          "source" : "40766f9915552038.json",
          "type" : "application/json",
          "size" : 2
        }, {
          "uid" : "a6410c3d753c631e",
          "name" : "Response headers 200",
          "source" : "a6410c3d753c631e.json",
          "type" : "application/json",
          "size" : 490
        } ],
        "parameters" : [ ],
        "stepsCount" : 0,
        "attachmentsCount" : 3,
        "shouldDisplayMessage" : false,
        "hasContent" : true
      }, {
        "name" : "GET /api/spends/all?filterCurrency=USD",
        "time" : {
          "start" : 1750753066834,
          "stop" : 1750753066857,
          "duration" : 23
        },
        "status" : "passed",
        "steps" : [ ],
        "attachments" : [ {
          "uid" : "dc6880442fc96f77",
          "name" : "Request",
          "source" : "dc6880442fc96f77.html",
          "type" : "text/html",
          "size" : 3623
        }, {
          "uid" : "be3cde7ad04c0dd0",
          "name" : "Response json 200",
          "source" : "be3cde7ad04c0dd0.json",
          "type" : "application/json",
          "size" : 2
        }, {
          "uid" : "d1ee09e47069d354",
          "name" : "Response headers 200",
          "source" : "d1ee09e47069d354.json",
          "type" : "application/json",
          "size" : 490
        } ],
        "parameters" : [ ],
        "stepsCount" : 0,
        "attachmentsCount" : 3,
        "shouldDisplayMessage" : false,
        "hasContent" : true
      }, {
        "name" : "GET /api/spends/all?filterCurrency=EUR",
        "time" : {
          "start" : 1750753066866,
          "stop" : 1750753066884,
          "duration" : 18
        },
        "status" : "passed",
        "steps" : [ ],
        "attachments" : [ {
          "uid" : "13e08e6faced3bc6",
          "name" : "Request",
          "source" : "13e08e6faced3bc6.html",
          "type" : "text/html",
          "size" : 3623
        }, {
          "uid" : "6df6d5eb97ad09d8",
          "name" : "Response json 200",
          "source" : "6df6d5eb97ad09d8.json",
          "type" : "application/json",
          "size" : 2
        }, {
          "uid" : "c48fafcac08c61db",
          "name" : "Response headers 200",
          "source" : "c48fafcac08c61db.json",
          "type" : "application/json",
          "size" : 490
        } ],
        "parameters" : [ ],
        "stepsCount" : 0,
        "attachmentsCount" : 3,
        "shouldDisplayMessage" : false,
        "hasContent" : true
      } ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 4,
      "attachmentsCount" : 12,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    }, {
      "name" : "Получить id всех категорий пользователя",
      "time" : {
        "start" : 1750753066884,
        "stop" : 1750753066907,
        "duration" : 23
      },
      "status" : "passed",
      "steps" : [ {
        "name" : "GET /api/categories/all",
        "time" : {
          "start" : 1750753066896,
          "stop" : 1750753066907,
          "duration" : 11
        },
        "status" : "passed",
        "steps" : [ ],
        "attachments" : [ {
          "uid" : "e3ad4533aeb4dff0",
          "name" : "Request",
          "source" : "e3ad4533aeb4dff0.html",
          "type" : "text/html",
          "size" : 3623
        }, {
          "uid" : "29671954eb389f7e",
          "name" : "Response json 200",
          "source" : "29671954eb389f7e.json",
          "type" : "application/json",
          "size" : 150
        }, {
          "uid" : "58a07574ede2b5ae",
          "name" : "Response headers 200",
          "source" : "58a07574ede2b5ae.json",
          "type" : "application/json",
          "size" : 490
        } ],
        "parameters" : [ ],
        "stepsCount" : 0,
        "attachmentsCount" : 3,
        "shouldDisplayMessage" : false,
        "hasContent" : true
      } ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "exclude_archived",
        "value" : "False"
      } ],
      "stepsCount" : 1,
      "attachmentsCount" : 3,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    }, {
      "name" : "Удалить из БД категории из списка",
      "time" : {
        "start" : 1750753066907,
        "stop" : 1750753066982,
        "duration" : 75
      },
      "status" : "broken",
      "statusMessage" : "sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nDETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\n\n[SQL: DELETE FROM category WHERE category.id = %(id)s]\n[parameters: {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/allure_commons/_allure.py\", line 202, in impl\n    return func(*a, **kw)\n  File \"/home/runner/work/niffler-py-st3/niffler-py-st3/python_test/databases/spend_db.py\", line 57, in delete_categories_by_ids\n    self.delete_category(category_id)\n    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/allure_commons/_allure.py\", line 202, in impl\n    return func(*a, **kw)\n  File \"/home/runner/work/niffler-py-st3/niffler-py-st3/python_test/databases/spend_db.py\", line 52, in delete_category\n    session.commit()\n    ~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 2032, in commit\n    trans.commit(_to_root=True)\n    ~~~~~~~~~~~~^^^^^^^^^^^^^^^\n  File \"<string>\", line 2, in commit\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py\", line 139, in _go\n    ret_value = fn(self, *arg, **kw)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 1313, in commit\n    self._prepare_impl()\n    ~~~~~~~~~~~~~~~~~~^^\n  File \"<string>\", line 2, in _prepare_impl\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py\", line 139, in _go\n    ret_value = fn(self, *arg, **kw)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 1288, in _prepare_impl\n    self.session.flush()\n    ~~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4353, in flush\n    self._flush(objects)\n    ~~~~~~~~~~~^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4488, in _flush\n    with util.safe_reraise():\n         ~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/util/langhelpers.py\", line 146, in __exit__\n    raise exc_value.with_traceback(exc_tb)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4449, in _flush\n    flush_context.execute()\n    ~~~~~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py\", line 466, in execute\n    rec.execute(self)\n    ~~~~~~~~~~~^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py\", line 679, in execute\n    util.preloaded.orm_persistence.delete_obj(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        self.mapper,\n        ^^^^^^^^^^^^\n        uow.states_for_mapper_hierarchy(self.mapper, True, False),\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n        uow,\n        ^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py\", line 193, in delete_obj\n    _emit_delete_statements(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        base_mapper,\n        ^^^^^^^^^^^^\n    ...<3 lines>...\n        delete,\n        ^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py\", line 1465, in _emit_delete_statements\n    c = connection.execute(\n        statement, del_objects, execution_options=execution_options\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1416, in execute\n    return meth(\n        self,\n        distilled_parameters,\n        execution_options or NO_OPTIONS,\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/sql/elements.py\", line 523, in _execute_on_connection\n    return connection._execute_clauseelement(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        self, distilled_params, execution_options\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1638, in _execute_clauseelement\n    ret = self._execute_context(\n        dialect,\n    ...<8 lines>...\n        cache_hit=cache_hit,\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1843, in _execute_context\n    return self._exec_single_context(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~^\n        dialect, context, statement, parameters\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1983, in _exec_single_context\n    self._handle_dbapi_exception(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        e, str_statement, effective_parameters, cursor, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 2352, in _handle_dbapi_exception\n    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1964, in _exec_single_context\n    self.dialect.do_execute(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        cursor, str_statement, effective_parameters, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/default.py\", line 945, in do_execute\n    cursor.execute(statement, parameters)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^\n",
      "steps" : [ {
        "name" : "Удалить из БД категорию",
        "time" : {
          "start" : 1750753066907,
          "stop" : 1750753066972,
          "duration" : 65
        },
        "status" : "broken",
        "statusMessage" : "sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nDETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\n\n[SQL: DELETE FROM category WHERE category.id = %(id)s]\n[parameters: {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)\n",
        "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/allure_commons/_allure.py\", line 202, in impl\n    return func(*a, **kw)\n  File \"/home/runner/work/niffler-py-st3/niffler-py-st3/python_test/databases/spend_db.py\", line 52, in delete_category\n    session.commit()\n    ~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 2032, in commit\n    trans.commit(_to_root=True)\n    ~~~~~~~~~~~~^^^^^^^^^^^^^^^\n  File \"<string>\", line 2, in commit\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py\", line 139, in _go\n    ret_value = fn(self, *arg, **kw)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 1313, in commit\n    self._prepare_impl()\n    ~~~~~~~~~~~~~~~~~~^^\n  File \"<string>\", line 2, in _prepare_impl\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py\", line 139, in _go\n    ret_value = fn(self, *arg, **kw)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 1288, in _prepare_impl\n    self.session.flush()\n    ~~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4353, in flush\n    self._flush(objects)\n    ~~~~~~~~~~~^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4488, in _flush\n    with util.safe_reraise():\n         ~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/util/langhelpers.py\", line 146, in __exit__\n    raise exc_value.with_traceback(exc_tb)\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py\", line 4449, in _flush\n    flush_context.execute()\n    ~~~~~~~~~~~~~~~~~~~~~^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py\", line 466, in execute\n    rec.execute(self)\n    ~~~~~~~~~~~^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py\", line 679, in execute\n    util.preloaded.orm_persistence.delete_obj(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        self.mapper,\n        ^^^^^^^^^^^^\n        uow.states_for_mapper_hierarchy(self.mapper, True, False),\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n        uow,\n        ^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py\", line 193, in delete_obj\n    _emit_delete_statements(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        base_mapper,\n        ^^^^^^^^^^^^\n    ...<3 lines>...\n        delete,\n        ^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py\", line 1465, in _emit_delete_statements\n    c = connection.execute(\n        statement, del_objects, execution_options=execution_options\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1416, in execute\n    return meth(\n        self,\n        distilled_parameters,\n        execution_options or NO_OPTIONS,\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/sql/elements.py\", line 523, in _execute_on_connection\n    return connection._execute_clauseelement(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        self, distilled_params, execution_options\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1638, in _execute_clauseelement\n    ret = self._execute_context(\n        dialect,\n    ...<8 lines>...\n        cache_hit=cache_hit,\n    )\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1843, in _execute_context\n    return self._exec_single_context(\n           ~~~~~~~~~~~~~~~~~~~~~~~~~^\n        dialect, context, statement, parameters\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1983, in _exec_single_context\n    self._handle_dbapi_exception(\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^\n        e, str_statement, effective_parameters, cursor, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 2352, in _handle_dbapi_exception\n    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py\", line 1964, in _exec_single_context\n    self.dialect.do_execute(\n    ~~~~~~~~~~~~~~~~~~~~~~~^\n        cursor, str_statement, effective_parameters, context\n        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n    )\n    ^\n  File \"/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/default.py\", line 945, in do_execute\n    cursor.execute(statement, parameters)\n    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^\n",
        "steps" : [ ],
        "attachments" : [ {
          "uid" : "c6c646fc3bcc2f91",
          "name" : "SELECT niffler-spend",
          "source" : "c6c646fc3bcc2f91.txt",
          "type" : "text/plain",
          "size" : 218
        }, {
          "uid" : "877eb9877e05c58a",
          "name" : "DELETE niffler-spend",
          "source" : "877eb9877e05c58a.txt",
          "type" : "text/plain",
          "size" : 77
        } ],
        "parameters" : [ {
          "name" : "category_id",
          "value" : "'f6f89ac6-d807-4f96-91fd-984e72abfd62'"
        } ],
        "stepsCount" : 0,
        "attachmentsCount" : 2,
        "shouldDisplayMessage" : true,
        "hasContent" : true
      } ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "categories_ids",
        "value" : "['f6f89ac6-d807-4f96-91fd-984e72abfd62']"
      } ],
      "stepsCount" : 1,
      "attachmentsCount" : 2,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 9,
    "attachmentsCount" : 17,
    "shouldDisplayMessage" : false,
    "hasContent" : true
  }, {
    "name" : "[S] Worker Id",
    "time" : {
      "start" : 1750753066546,
      "stop" : 1750753066546,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "[S]  Session Faker",
    "time" : {
      "start" : 1750753066027,
      "stop" : 1750753066030,
      "duration" : 3
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "[S] Spends Client",
    "time" : {
      "start" : 1750753066545,
      "stop" : 1750753066545,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "[S] Spend Db",
    "time" : {
      "start" : 1750753066546,
      "stop" : 1750753066659,
      "duration" : 113
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "status" : "broken",
    "statusMessage" : "sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nDETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\n\n[SQL: DELETE FROM category WHERE category.id = %(id)s]\n[parameters: {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)",
    "statusTrace" : "self = <sqlalchemy.engine.base.Connection object at 0x7fd71cfbf230>\ndialect = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7fd71cfbd7f0>\ncontext = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7fd71ce916e0>\nstatement = <sqlalchemy.dialects.postgresql.base.PGCompiler object at 0x7fd71ceb5bd0>\nparameters = [{'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\n\n    def _exec_single_context(\n        self,\n        dialect: Dialect,\n        context: ExecutionContext,\n        statement: Union[str, Compiled],\n        parameters: Optional[_AnyMultiExecuteParams],\n    ) -> CursorResult[Any]:\n        \"\"\"continue the _execute_context() method for a single DBAPI\n        cursor.execute() or cursor.executemany() call.\n    \n        \"\"\"\n        if dialect.bind_typing is BindTyping.SETINPUTSIZES:\n            generic_setinputsizes = context._prepare_set_input_sizes()\n    \n            if generic_setinputsizes:\n                try:\n                    dialect.do_set_input_sizes(\n                        context.cursor, generic_setinputsizes, context\n                    )\n                except BaseException as e:\n                    self._handle_dbapi_exception(\n                        e, str(statement), parameters, None, context\n                    )\n    \n        cursor, str_statement, parameters = (\n            context.cursor,\n            context.statement,\n            context.parameters,\n        )\n    \n        effective_parameters: Optional[_AnyExecuteParams]\n    \n        if not context.executemany:\n            effective_parameters = parameters[0]\n        else:\n            effective_parameters = parameters\n    \n        if self._has_events or self.engine._has_events:\n            for fn in self.dispatch.before_cursor_execute:\n                str_statement, effective_parameters = fn(\n                    self,\n                    cursor,\n                    str_statement,\n                    effective_parameters,\n                    context,\n                    context.executemany,\n                )\n    \n        if self._echo:\n            self._log_info(str_statement)\n    \n            stats = context._get_cache_stats()\n    \n            if not self.engine.hide_parameters:\n                self._log_info(\n                    \"[%s] %r\",\n                    stats,\n                    sql_util._repr_params(\n                        effective_parameters,\n                        batches=10,\n                        ismulti=context.executemany,\n                    ),\n                )\n            else:\n                self._log_info(\n                    \"[%s] [SQL parameters hidden due to hide_parameters=True]\",\n                    stats,\n                )\n    \n        evt_handled: bool = False\n        try:\n            if context.execute_style is ExecuteStyle.EXECUTEMANY:\n                effective_parameters = cast(\n                    \"_CoreMultiExecuteParams\", effective_parameters\n                )\n                if self.dialect._has_events:\n                    for fn in self.dialect.dispatch.do_executemany:\n                        if fn(\n                            cursor,\n                            str_statement,\n                            effective_parameters,\n                            context,\n                        ):\n                            evt_handled = True\n                            break\n                if not evt_handled:\n                    self.dialect.do_executemany(\n                        cursor,\n                        str_statement,\n                        effective_parameters,\n                        context,\n                    )\n            elif not effective_parameters and context.no_parameters:\n                if self.dialect._has_events:\n                    for fn in self.dialect.dispatch.do_execute_no_params:\n                        if fn(cursor, str_statement, context):\n                            evt_handled = True\n                            break\n                if not evt_handled:\n                    self.dialect.do_execute_no_params(\n                        cursor, str_statement, context\n                    )\n            else:\n                effective_parameters = cast(\n                    \"_CoreSingleExecuteParams\", effective_parameters\n                )\n                if self.dialect._has_events:\n                    for fn in self.dialect.dispatch.do_execute:\n                        if fn(\n                            cursor,\n                            str_statement,\n                            effective_parameters,\n                            context,\n                        ):\n                            evt_handled = True\n                            break\n                if not evt_handled:\n>                   self.dialect.do_execute(\n                        cursor, str_statement, effective_parameters, context\n                    )\n\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1964: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7fd71cfbd7f0>\ncursor = <cursor object at 0x7fd71c8a09a0; closed: -1>\nstatement = 'DELETE FROM category WHERE category.id = %(id)s'\nparameters = {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}\ncontext = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7fd71ce916e0>\n\n    def do_execute(self, cursor, statement, parameters, context=None):\n>       cursor.execute(statement, parameters)\nE       psycopg2.errors.ForeignKeyViolation: update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nE       DETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\n\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/default.py:945: ForeignKeyViolation\n\nThe above exception was the direct cause of the following exception:\n\ntmp_path_factory = TempPathFactory(_given_basetemp=PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw2'), _trace=<pluggy._tracing.TagTrac...4101490>, _basetemp=PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw2'), _retention_count=3, _retention_policy='all')\nworker_id = 'gw2'\nspends_client = <python_test.data_helper.api_helper.SpendsHttpClient object at 0x7fd71d0a5d30>\nspend_db = <python_test.databases.spend_db.SpendDb object at 0x7fd71d0a5160>\n\n    @pytest.fixture(scope=\"module\", autouse=True)\n    def module_fixture(tmp_path_factory, worker_id, spends_client: SpendsHttpClient, spend_db: SpendDb):\n        def _prepare_state():\n            all_spends_ids = spends_client.get_ids_all_spending()\n            for spend_id in all_spends_ids:\n                spends_client.delete_spending_by_id(spend_id)\n            categories_ids = spends_client.get_ids_all_categories()\n            spend_db.delete_categories_by_ids(categories_ids)\n    \n        if worker_id == \"master\":\n            _prepare_state()\n    \n        root_tmp_dir = tmp_path_factory.getbasetemp().parent\n    \n        fn = root_tmp_dir / \"prepare\"\n        with FileLock(str(fn) + \".lock\"):\n            if fn.is_file():\n                pass\n            else:\n>               _prepare_state()\n\ntest/test_api.py:40: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ntest/test_api.py:28: in _prepare_state\n    spend_db.delete_categories_by_ids(categories_ids)\ndatabases/spend_db.py:57: in delete_categories_by_ids\n    self.delete_category(category_id)\ndatabases/spend_db.py:52: in delete_category\n    session.commit()\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:2032: in commit\n    trans.commit(_to_root=True)\n<string>:2: in commit\n    ???\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py:139: in _go\n    ret_value = fn(self, *arg, **kw)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:1313: in commit\n    self._prepare_impl()\n<string>:2: in _prepare_impl\n    ???\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/state_changes.py:139: in _go\n    ret_value = fn(self, *arg, **kw)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:1288: in _prepare_impl\n    self.session.flush()\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:4353: in flush\n    self._flush(objects)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:4488: in _flush\n    with util.safe_reraise():\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/util/langhelpers.py:146: in __exit__\n    raise exc_value.with_traceback(exc_tb)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/session.py:4449: in _flush\n    flush_context.execute()\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py:466: in execute\n    rec.execute(self)\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/unitofwork.py:679: in execute\n    util.preloaded.orm_persistence.delete_obj(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py:193: in delete_obj\n    _emit_delete_statements(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/orm/persistence.py:1465: in _emit_delete_statements\n    c = connection.execute(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1416: in execute\n    return meth(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/sql/elements.py:523: in _execute_on_connection\n    return connection._execute_clauseelement(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1638: in _execute_clauseelement\n    ret = self._execute_context(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1843: in _execute_context\n    return self._exec_single_context(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1983: in _exec_single_context\n    self._handle_dbapi_exception(\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:2352: in _handle_dbapi_exception\n    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/base.py:1964: in _exec_single_context\n    self.dialect.do_execute(\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7fd71cfbd7f0>\ncursor = <cursor object at 0x7fd71c8a09a0; closed: -1>\nstatement = 'DELETE FROM category WHERE category.id = %(id)s'\nparameters = {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}\ncontext = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7fd71ce916e0>\n\n    def do_execute(self, cursor, statement, parameters, context=None):\n>       cursor.execute(statement, parameters)\nE       sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nE       DETAIL:  Key (id)=(f6f89ac6-d807-4f96-91fd-984e72abfd62) is still referenced from table \"spend\".\nE       \nE       [SQL: DELETE FROM category WHERE category.id = %(id)s]\nE       [parameters: {'id': UUID('f6f89ac6-d807-4f96-91fd-984e72abfd62')}]\nE       (Background on this error at: https://sqlalche.me/e/20/gkpj)\n\n/opt/hostedtoolcache/Python/3.13.5/x64/lib/python3.13/site-packages/sqlalchemy/engine/default.py:945: IntegrityError",
    "steps" : [ ],
    "attachments" : [ {
      "uid" : "856f58c9a836f41a",
      "name" : "log",
      "source" : "856f58c9a836f41a.txt",
      "type" : "text/plain",
      "size" : 84
    } ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 1,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ ],
  "labels" : [ {
    "name" : "story",
    "value" : "Категории"
  }, {
    "name" : "feature",
    "value" : "API"
  }, {
    "name" : "epic",
    "value" : "Приложение Niffler"
  }, {
    "name" : "host",
    "value" : "fv-az1692-520"
  }, {
    "name" : "thread",
    "value" : "7589-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "test.test_api"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "count",
    "value" : "50"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 3,
        "skipped" : 0,
        "passed" : 0,
        "unknown" : 0,
        "total" : 3
      },
      "items" : [ {
        "uid" : "2c4461dcefa46905",
        "reportUrl" : "https://shade1471.github.io/niffler-py-st3/43/#testresult/2c4461dcefa46905",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.ConnectionError: HTTPConnectionPool(host='frontend.niffler.dc', port=80): Max retries exceeded with url: /authorized?code=f_7ZaLs_ugxv2_KBSVnEK0-o0EuhzJAiVxQr6zvsl2rNvJFWrNrcXo1awgLKtndla6m4Cphq23tEaFRXEQPN39vOjXwc5JB7_psbZfX0ckkXm6vkvTYRGtsHgk_IpySZ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff2a82cc190>: Failed to establish a new connection: [Errno 111] Connection refused'))",
        "time" : {
          "start" : 1750752663404,
          "stop" : 1750752663404,
          "duration" : 0
        }
      }, {
        "uid" : "44e9d9ca10838965",
        "reportUrl" : "https://shade1471.github.io/niffler-py-st3/42/#testresult/44e9d9ca10838965",
        "status" : "broken",
        "statusDetails" : "sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table \"category\" violates foreign key constraint \"fk_spend_category\" on table \"spend\"\nDETAIL:  Key (id)=(7a958c45-f242-4652-93b6-36e6c5403f6d) is still referenced from table \"spend\".\n\n[SQL: DELETE FROM category WHERE category.id = %(id)s]\n[parameters: {'id': UUID('7a958c45-f242-4652-93b6-36e6c5403f6d')}]\n(Background on this error at: https://sqlalche.me/e/20/gkpj)",
        "time" : {
          "start" : 1750735602144,
          "stop" : 1750735602144,
          "duration" : 0
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "27bf3d3363484158.json",
  "parameterValues" : [ "50" ]
}